### ISSUE TITLE
Checkstyle config enforces line break at "@Annotation\n|package foo;"

### ISSUE THREAD

**Issue by dflemstr** [spotify/spotify-checkstyle-config#13]
The config rejects this code (in a `package-info.java`):

```java
@ParametersAreNonnullByDefault
package com.spotify.myservice;
```

It accepts the following code:

```java
@ParametersAreNonnullByDefault

package com.spotify.myservice;
```

The Google style guide doesn't say anything about this but I think the second code looks worse and the [automatic formatter](https://github.com/google/google-java-format) removes the empty line if you add it.

Opening this as an Issue as I'm not familiar with the Checkstyle config syntax so I don't know the idiomatic way to fix it, and I want to give people a chance to bikeshed about this ;)

**Comment by mattnworb** [comment]
Doesn't the checkstyle warning print out the name of the check that failed for the first example? Knowing which check it is would make it a lot easier to address.

Tangental but we should probably add [PackageAnnotation](http://checkstyle.sourceforge.net/config_annotation.html#PackageAnnotation) too to enforce package annotations are only in `package-info.java` files

**Comment by dflemstr** [comment]
@mattnworb Right, sorry about not mentioning the check:

```text
[WARN] xyz.java:2: 'package' should be separated from previous statement. [EmptyLineSeparator]
```

**Comment by mattnworb** [comment]
I tried to reproduce this in https://github.com/mattnworb/spotify-checkstyle-config/commit/beb4adcdec35dc499afffee8cb5c4dd12c9bd96a and for some reason the check doesn't fail for me in that case. Is my test case different than the actual source you have?

**Comment by mattnworb** [comment]
seems related: https://github.com/checkstyle/checkstyle/issues/3426

**Comment by mattnworb** [comment]
the fix for that issue is in checkstyle:7.4 btw, although it doesn't explain why my test case doesn't generate a warning

**Comment by dflemstr** [comment]
I suppose we should upgrade checkstyle in our internal root POM then; we seem to be using 6.19 at the moment. Will file an issue there.
